test/test.js

```diff
@@ -5,259 +5,162 @@ var decimalDevider = 100000000;
-        }).then(function(icoBalance) {
-            // save initial ICO balance
-            initialICOBalance = icoBalance.toNumber();
-            // fix 5 ETH investment using fundBTC()
-            return icoContract.fixInvestment(icoInvestor, // beneficiary
-                web3.toWei(5, "Ether"), // amount
-                Date.now() / 1000, // time of investment (in seconds)
-                {
-                    from: icoOwner, // only owner can call this function
-                    gas: gasAmount
-                });
-        }).then(function(tx_id) {
-            return tokenContract.balanceOf.call(icoInvestor);
+            return instance.allocationAddressICO.call();
+        }).then(function(address) {
+            assert.equal(address, allocationAddress, "Token Contract Allocation address should be equal to 0x1111111111111111111111111111111111111111");
+        }).then(done);
+    });
+
+    it("Should verify founders balance", function(done) {
+        HumaniqToken.deployed().then(function(instance) {
+            return instance.balanceOf.call(multisig);
         }).then(function(balance) {
-            // check that beneficiary received correct number of tokens
+            assert.equal(balance.toNumber(), foundersBalance, "Wrong founders balance");
+        }).then(done);
+    });
 
-            assert.closeTo(balance.toNumber() / decimalDevider,
-                initialTokens + (web3.toWei(5, "Ether") / baseTokenPrice) * bonus,
-                0.0000001, // possible javascript computational error
-                "Wrong number of tokens was given");
-            console.log(balance.toNumber() / decimalDevider - initialTokens, "tokens were issued for 5 Ether");
-            return tokenContract.totalSupply.call();
-        }).then(function(totalSupply) {
-            // check that totalSupply of tokens is correct
-            assert.closeTo(totalSupply.toNumber() / decimalDevider,
-                initialTokenSupply + (web3.toWei(5, "Ether") / baseTokenPrice) * bonus,
-                0.0000001, // possible javascript computational error
-                "Wrong total supply");
-            return icoContract.icoBalance.call();
-        }).then(function(icoBalance) {
-            // check that ICO balance is correct
-            assert.equal(icoBalance.toNumber(),
-                initialICOBalance + parseInt(web3.toWei(5, "Ether")),
-                "Wrong ICO balance");
+    it("Should verify max total supply", function(done) {
+        HumaniqToken.deployed().then(function(instance) {
+            return instance.maxTotalSupply.call();
+        }).then(function(supply) {
+            assert.equal(supply.toNumber(), 5 * (ICOsupply + foundersBalance + preICOsupply), "Wrong max total supply");
         }).then(done);
     });
 
-    it("Should return current bonus", function(done) {
+    it("Should check bonuses", function(done) {
+
+        var dateBeforeICO = startICODate - 1;
```

Comment:
Should we iterate over every date and check the bonus at every timestamp with a minute step?